test/algorithms/afl_symcc/parallel.cpp

```diff
@@ -0,0 +1,131 @@
+    std::vector<char> input{17, 0, 0, 0, 3, 4, 1, 0};
+    FUZZUF_SETUP_SINGLE_INITIAL_INPUT(input)
+  }
+
+  BOOST_TEST_CHECKPOINT("initialized dirs");
+  const char *argv[] = {"fuzzuf",
+                        "afl_symcc",
+                        "-i",
+                        input_dir.c_str(),
+                        "-o",
+                        output_dir.c_str(),
+                        "-M",
+                        "fuzzer0",
+                        "--symcc_target",
+                        TEST_BINARY_DIR "/put/symcc/symcc-hard_to_branch",
+                        TEST_BINARY_DIR "/put/afl_gcc/afl_gcc-hard_to_branch",
+                        nullptr};
+  constexpr int argc = 11;
+  auto fuzzer = fuzzuf::cli::CreateFuzzerInstanceFromArgv(argc, argv);
+
+  BOOST_TEST_CHECKPOINT("created fuzzer");
+
+  // Minimum required cycles to invoke synchronization
+  for (size_t i = 0u; i != 2u; ++i) {
+    fuzzer->OneLoop();
+  }
+
+  // At least one crash input is produced
+  BOOST_CHECK_GE(
+      fuzzuf::utils::CountRegularFiles(output_dir / "fuzzer0" / "crashes"), 1);
+
+  BOOST_TEST_CHECKPOINT("done");
+}
+
+BOOST_AUTO_TEST_CASE(ExecuteSymCCWithoutParallel) {
+  // Setup root directory
+  // NOLINTBEGIN(cppcoreguidelines-pro-type-cstyle-cast,cppcoreguidelines-pro-type-member-init,cppcoreguidelines-special-member-functions,hicpp-explicit-conversions)
+  FUZZUF_STANDARD_TEST_DIRS
+  // NOLINTEND(cppcoreguidelines-pro-type-cstyle-cast,cppcoreguidelines-pro-type-member-init,cppcoreguidelines-special-member-functions,hicpp-explicit-conversions)
+
+  BOOST_TEST_CHECKPOINT("before init state");
+
+  {
+    // This input value is close to crash input so the crash input will be
+    // discovered for small number of cycles
+    std::vector<char> input{17, 0, 0, 0, 3, 4, 1, 0};
+    FUZZUF_SETUP_SINGLE_INITIAL_INPUT(input)
+  }
+
+  BOOST_TEST_CHECKPOINT("initialized dirs");
+  const char *argv[] = {"fuzzuf",
+                        "afl_symcc",
+                        "-i",
+                        input_dir.c_str(),
+                        "-o",
+                        output_dir.c_str(),
+                        "-M",
+                        "fuzzer0",
+                        "--symcc_target",
+                        TEST_BINARY_DIR "/put/symcc/symcc-hard_to_branch",
+                        TEST_BINARY_DIR "/put/afl_gcc/afl_gcc-hard_to_branch",
+                        nullptr};
+  constexpr int argc = 11;
+  auto fuzzer = fuzzuf::cli::CreateFuzzerInstanceFromArgv(argc, argv);
+
+  BOOST_TEST_CHECKPOINT("created fuzzer");
+
+  // Minimum required cycles to invoke synchronization
+  for (size_t i = 0u; i != 2u; ++i) {
+    fuzzer->OneLoop();
+  }
+
+  std::size_t crash_count = 0u;
+  for (const auto &e :
+       fs::directory_iterator(output_dir / "fuzzer0" / "crashes")) {
+#ifdef HAS_CXX_STD_FILESYSTEM
+    BOOST_CHECK(e.is_regular_file());
+#else
+    BOOST_CHECK(fs::is_regular_file(e.path()));
+#endif
+    ++crash_count;
+  }
```

Comment:
It looks like the created crash_count value is not used in the test. Add comments on what needs to be tested.